### ISSUE TITLE
output.path configuration property has no effect

### ISSUE THREAD

**Issue by bstro** [ethul/purs-loader#4]
I'm attempting to start using the purescript loader using your example as a starting point. However, changing 'filename' appears to have no effect… the file is simply compiled to the output directory in a directory named after the module itself, as index.js

Is this intended? I can omit the filename property with no apparent change in behavior. While its compiling, its a little unsettling.

```output: {
    path: path.join(__dirname, 'output'),
    filename: 'app.js'
}, ```

**Comment by ethul** [comment]
The way that the purescript file is compile is such that it creates a commonjs module, which is the behaviour you describe. A directory is created with the name of the purescript module, and inside is an `index.js` file containing the compiled JS.

When you specify the `output` in your webpack config, this should determine the path and filename of the resulting webpack bundle when you run the webpack command.

Also, as a heads up, I am on the verge of deprecating the `purs-loader`. Previously, I thought I had a use-case for this. However, I don't know if this type of loader makes sense anymore. The reason is that the loader will compile each purescript file independently. I think it makes sense to compile all the purescript files up-front, then require them via webpack. The `purs-loader` is not necessary to achieve this.

If you have a use-case for the `purs-loader`, I'd be happy to chat more about the best direction to take with this loader.

**Comment by antonycourtney** [comment]
Thanks very much for the example.  The entry.js boilerplate was the key that I was missing.

The next issue I've encountered is that this doesn't seem to find any purescript libraries installed using bower, and there doesn't appear to be any to set a path variable to get psc-make to search for them.
I'm specifically trying to use Data.Maybe, installed from bower package purescript-maybe.

Thanks for the heads up that you are considering deprecating it.  Totally understand but I'll be sorry to see it go.  In my JavaScript work I moved from grunt to gulp to gulp+browserify to webpack.  I couldn't certainly go back to using grunt, as that seems to be what's best supported / encouraged in the PureScript community, but doing so feels like something of a step backwards.

**Comment by ethul** [comment]
Glad that the example helped.

Regarding the issue with purescript libraries installed with bower, I have a tentative solution in at branch [topic/bower-components](https://github.com/ethul/purs-loader/tree/topic/bower-components) (please see the updated example). I don't know if this is ultimately the best approach, but it might work for your use-case. As an alternative, we could make the glob pattern a query parameter instead.

I am still on the fence about this library. I'd like to keep it if we can figure out the best use-case for it. It might actually be something to run with if we can iron these issues, which would be great!

**Comment by antonycourtney** [comment]
Wow, awesome, thank you!  I will give it a try soon.

For what it's worth, after fiddling around I've ended up with a setup I'm reasonably happy with.  The main reason I wanted webpack was for --watch because gulp.watch wasn't working well (it would just exit if there were compilation errors).  I eventually found this issue: https://github.com/deepak1556/gulp-browserify/issues/29 and now have a gulp setup using gulp-plumber that makes gulp.watch work correctly, even if there are purescript compile errors.

It seems that psc is effectively building a single JS bundle so somewhat subsumes the role of webpack or browserify....I take it that's why you're on the fence about this project?

Anyway, thanks for all the help and effort here.  Happy to share my gulp setup if you are interested.

**Comment by ethul** [comment]
I've published version `0.0.3` of `purs-loader` to include your suggestion for the example and the fix for handling the purescript libraries. I am thinking that a good use-case might be the one you mention, where one only wants to use webpack, and not use it in conjunction with gulp or grunt. In any case, I will keep the loader, but I am open to moving it in another direction if the need arises.

Thank you for your feedback and help in improving this loader.
